src/tqdm_publisher/_demos/_parallel_bars/_server.py

```diff
@@ -127,13 +125,13 @@ def run_parallel_processes(*, all_task_times: List[List[float]], request_id: str
 
         total_tasks_iterable = as_completed(futures)
         total_tasks_progress_bar = TQDMProgressPublisher(
-            iterable=total_tasks_iterable, total=len(all_task_times), desc="Total tasks completed"
+            iterable=total_tasks_iterable, total=len(all_task_times), desc=f"Total tasks completed for {request_id}"
```

Comment:
@garrettmflynn The frontend display of the subprogress bars looks great!

Can you also add the same UI for the 'main' (top) progress bar too? One test is that this demo should use the custom `desc` seen here

Reply:
Technically this was already happening—but all bar information is only popupated on the _first_ update—and the main progress bar takes a while (i.e. until one of the sub-bars finishes).

I've updated the `TQDMProgressPublisher` to send the current state immediately after subscribing? Is that alright or do we want to handle that manually in each demo?

Reply:
Definitely better - could you move the `desc` and elapsed/remaining to under the main bar to stylistically match the others?